src/Dofs/DofHandler.jl

```diff
@@ -76,26 +78,73 @@ julia> vtk_point_data(vtkfile
 
 """
 
-# TODO: Make this immutable?
+const DEBUG = true
+
+@pure get_n_edges(ip::Interpolation) = get_n_edges(typeof(ip))
+@pure get_n_edgedofs(ip::Interpolation) = get_n_edgedofs(typeof(ip))
+@pure get_n_vertices(ip::Interpolation) = get_n_vertices(typeof(ip))
+@pure getedgelist(ip::Interpolation) = getedge(typeof(ip))
+@pure get_n_faces{order, refshape}(::Type{Interpolation{1, refshape, order}}) = 0
+@pure get_n_faces{order, refshape}(::Type{Interpolation{2, refshape, order}}) = 0
+
+@pure get_n_edges{order}(::Type{Lagrange{1, RefCube, order}}) = 1
+@pure get_n_edges{order}(::Type{Lagrange{2, RefCube, order}}) = 4
+@pure get_n_edges{order}(::Type{Lagrange{3, RefCube, order}}) = 12
+
+@pure get_n_vertices{order}(::Type{Lagrange{1, RefCube, order}}) = 2
+@pure get_n_vertices{order}(::Type{Lagrange{2, RefCube, order}}) = 4
+@pure get_n_vertices{order}(::Type{Lagrange{3, RefCube, order}}) = 8
+
+@pure get_n_faces(ip::Interpolation) = get_n_faces(typeof(ip))
+
+@pure get_n_faces{order}(::Type{Lagrange{3, RefCube, order}}) = 6
+
+@pure get_n_cells(ip::Interpolation) = get_n_cells(typeof(ip))
+@pure get_n_cells{dim, order}(::Type{Lagrange{dim, RefCube, order}}) = 1
+
+@pure getedgelist{order}(::Type{Lagrange{1, RefCube, order}}) = ((1,), (2,))
+@pure getedgelist{order}(::Type{Lagrange{2, RefCube, order}}) = ((1,2), (2, 3), (3,4), (4, 1))
+
+# Computes the number of degrees of freedom that sits in every edge for a SINGLE edge
+@pure get_n_edgedofs{dim, order}(T::Type{Lagrange{dim, RefCube, order}}) = (order - 1)
+
+@pure get_n_celldofs(ip::Interpolation) = get_n_celldofs(typeof(ip))
+@pure get_n_celldofs{dim, order}(T::Type{Lagrange{dim, RefCube, order}}) = (order - 1)^dim
+
+# DofHandler
 type DofHandler{dim, N, T, M}
-    dofs_nodes::Matrix{Int}
-    dofs_cells::Matrix{Int} # TODO <- Is this needed or just extract from dofs_nodes?
+    dofs_cells::Matrix{Int}
     field_names::Vector{Symbol}
+    interpolations::Vector{Interpolation}
     dof_dims::Vector{Int}
-    closed::ScalarWrapper{Bool}
-    dofs_vec::Vector{Int}
+    closed::JuAFEM.ScalarWrapper{Bool}
     grid::Grid{dim, N, T, M}
 end
 
-function DofHandler(m::Grid)
-    DofHandler(Matrix{Int}(0, 0), Matrix{Int}(0, 0), Symbol[], Int[], ScalarWrapper(false), Int[], m)
+function DofHandler(grid::Grid)
+    DofHandler(Matrix{Int}(0, 0), Symbol[], Interpolation[], Int[], JuAFEM.ScalarWrapper(false), grid)
+end
+
+isclosed(dh::DofHandler) = dh.closed[]
+
+
+# Add a field to the dofhandler ex `push!(dh, :u, 3)`
+function Base.push!(dh::DofHandler, name::Symbol, interpolation::Interpolation, dim::Int)
```

Comment:
This need to check that the interpolation is compatible with the cell in the grid (same dim and shape at least)